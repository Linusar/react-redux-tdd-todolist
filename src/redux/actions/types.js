const START_APP = 'START_APP';
const ADD_ELEMENT = 'ADD_ELEMENT';
const REMOVE_ELEMENT = 'REMOVE_ELEMENT';

export { START_APP, ADD_ELEMENT, REMOVE_ELEMENT };
