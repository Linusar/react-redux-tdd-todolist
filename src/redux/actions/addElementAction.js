import { createAction } from 'redux-actions';
import { ADD_ELEMENT } from './types';

const addElementAction = createAction(ADD_ELEMENT, task => task);

export default addElementAction;
