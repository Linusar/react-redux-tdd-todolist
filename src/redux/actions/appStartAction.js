import { createAction } from 'redux-actions';
import { START_APP } from './types';

const appStartAction = createAction(START_APP, status => status);

export default appStartAction;
