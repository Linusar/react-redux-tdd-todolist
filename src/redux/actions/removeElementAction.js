import { createAction } from 'redux-actions';
import { REMOVE_ELEMENT } from './types';

const removeElementAction = createAction(REMOVE_ELEMENT, id => id);

export default removeElementAction;
