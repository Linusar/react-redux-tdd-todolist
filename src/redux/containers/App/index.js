import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import * as ROUTES from '../../../constants/routes';
import HomeContainer from '../HomeContainer';

makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
}));

const App = () => (
  <Container component="main" maxWidth="xs">
    <Router>
      <Router>
        <Route exact path={ROUTES.LANDING} component={HomeContainer} />
      </Router>
    </Router>
  </Container>
);

export default App;
