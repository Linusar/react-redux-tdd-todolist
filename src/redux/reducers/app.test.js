/* global describe expect test */
import App from './app';
import { START_APP } from '../actions/types';

describe('INITIAL_STATE', () => {
  test('is correct', () => {
    const action = { type: 'dummy_action' };

    expect(App(undefined, action)).toMatchSnapshot();
  });
});

describe('START_APP', () => {
  test('returns the correct state', () => {
    const action = { type: START_APP, payload: 'status' };

    expect(App(undefined, action)).toMatchSnapshot();
  });
});
