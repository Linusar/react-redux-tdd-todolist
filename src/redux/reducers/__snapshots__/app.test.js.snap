// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE is correct 1`] = `
Object {
  "status": "App Runing!!!",
}
`;

exports[`START_APP returns the correct state 1`] = `
Object {
  "status": undefined,
}
`;
